src/group_theory/nilpotent.lean

```diff
@@ -208,48 +202,62 @@ begin
     exact eq_top_iff.mpr this }
 end
 
+lemma is_decending_rev_series_of_is_ascending
+  {H: ℕ → subgroup G} {n : ℕ} (hn : H n = ⊤) (hasc : is_ascending_central_series H) :
+  is_descending_central_series (λ (m : ℕ), H (n - m)) :=
+begin
+  cases hasc with h0 hH,
+  refine ⟨hn, λ x m hx g, _⟩,
+  dsimp at hx,
+  by_cases hm : n ≤ m,
+  { have hnm : n - m = 0 := tsub_eq_zero_iff_le.mpr hm,
+    rw [hnm, h0, subgroup.mem_bot] at hx,
+    subst hx,
+    convert subgroup.one_mem _,
+    group },
+  { push_neg at hm,
+    apply hH,
+    convert hx,
+    rw nat.sub_succ,
+    exact nat.succ_pred_eq_of_pos (tsub_pos_of_lt hm) },
+end
```

Comment:
This lemma and the following is just extracted from `nilpotent_iff_finite_descending_central_series`, because I need it below again, but not otherwise changed \</blame-deflector>